Upgrade `simpleaudio` to an updated fork that supports Python 3.12

jonathanperret opened the pull request:
## Problem

Fixes #725 (without breaking macOS audio, see #769).

## Proposed solution

We use a fork named `simpleaudio-patched` that is compatible with recent Python versions and has up-to-date binary builds.

## 🔬 Testing

Here's a test build that should be used to check audio works properly on all supported platforms: https://github.com/jonathanperret/ayab-desktop/releases/tag/1.0.1-simpleaudio2

I've tested that this release has correct audio in macOS 15.3.1.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **Chores**
	- Enhanced the macOS build process to ensure robust support for audio playback.
	- Updated the audio dependency to a patched version, providing improved performance and compatibility across macOS architectures.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dl1com commented:
I reverted PR #730 so we got a clean main branch again for now, so removal of the Qt sound code can be removed from this PR

jonathanperret commented:
> I reverted PR #730 so we got a clean main branch again for now, so removal of the Qt sound code can be removed from this PR

Right that's cleaner. I've rebased this PR and created a new test build.

VIPQualityPost commented:
Does this need testing on real hardware?
On Ubuntu 24.10 with Simulator all of the audio is working for me (start chime + row bell)
